### ISSUE TITLE
Tutorial Page hierarchy quickfix

### ISSUE THREAD

**Issue by lansingthomas** [atlas-engineer/nyxt#2837]
Dear Computer Titans,
On the _Tutorial_ page the headings and the subheadings have poor visual communication of hierarchy. 

I think the simple architecture is good and we can improve readability easily with some indentation.

Would someone please indent all the subheadings?

**Comment by lansingthomas** [comment]
Could this same hierarchy thing also be useful for the Help- system?

I find myself reading about Slots when I want to be reading about classes.

**Comment by lansingthomas** [comment]
great question @aadcg 

Yes, the body text as well. To improve skim-ability for readers. Thank you.

**Comment by lansingthomas** [comment]
-- Tutorial Heading
----### Tutorial Subheading 1
------nyxty nytum nyxtus nyxteroo nyxty nytum nyxtus nyxteroo
------nyxty nytum nyxtus nyxteroo
----### Tutorial Subheading 2
------nyxty nytum nyxtus nyxteroo nyxty nytum nyxtus nyxteroo
------nyxty nytum nyxtus nyxteroo

**Comment by jmercouris** [comment]
I think that would really help, I like it!

**Comment by aadcg** [comment]
I'm sorry but I have serious doubts about this one.

You're advocating for such a change in the tutorial alone, but I don't think that the tutorial differs greatly from the manual (besides length). So, I'd argue that it would need to be either both or none. Below I'll argue why I stand for none.

Our tutorial and manual are books that you read within Nyxt. Image an e-book that would be styled as you suggest. I don't think it helps in any meaningful way, it looks awful and wastes valuable space.

So, what does actually help readers to understand where they're are hierarchically? 

### Numbered titled sections

Not implemented. 

### Navigation keys to traverse the document 

Open the tutorial and hit the `?` key.  You'll find out that these keys allow you to navigate the document. These could be expanded.

![2023-03-15_15:22:34](https://user-images.githubusercontent.com/45483512/225321809-3efa0a63-d01e-4a7c-810d-e7b21dcbf2f3.png)

### A top bar that shows hierarchy

Not implemented.

Akin to programs like `info` do. Find an example below. Notice the two lines at the very top.

![2023-03-15_15:26:57](https://user-images.githubusercontent.com/45483512/225322511-31ab661d-de57-426f-84de-2b3b9e0eb1db.png)

-------

I'm not saying which ones we should pick. I'm just arguing that there are better ways to solve the problem at hand, and it would be a good idea to have a second round of thinking.

**Comment by aartaka** [comment]
> ### Numbered titled sections

Possible to do with `:nsection`, I believe.

> ### A top bar that shows hierarchy

Alternatively, a table of contents. I.e., `headings-panel` or #2848.

> I'm not saying which ones we should pick. I'm just arguing that there are better ways to solve the problem at hand, and it would be a good idea to have a second round of thinking.

I double this.

**Comment by aadcg** [comment]
> Alternatively, a table of contents. I.e., `headings-panel` or #2848.

Indeed, we already have `headings-panel`! So, from the list above, there only thing that is not in place are numbered sections. But I'm not saying we need them!

@lansingthomas, in light of these mechanisms we already have in place (that I might be unaware of), would you maintain that it's hard to follow the tutorial's hierarchy?

**Comment by lansingthomas** [comment]
Loving this conversation today -- 
Yes I am aware of the mechanisms in place (this time :P ) `headings-panel` (`t`) and keybindings to travel around `m` `n` `p`. They are very sensible for our emacs users and fun for all.

- these mechanisms get me where I need to go (turning the page to the right chapter)
- I would like a nicer reading experience once I arrive.

>You're advocating for such a change in the tutorial alone, but I don't think that the tutorial differs greatly from the manual (besides length). So, I'd argue that it would need to be either both or none. Below I'll argue why I stand for none.

Indentations should absolutely apply to the `manual` as well! 

#### Would look like this: 
---
![indentation mockup](https://user-images.githubusercontent.com/6666079/226990301-cab82257-6a4f-4b10-88ed-8889442cbafc.png)

- See how the headers and the accordion triangles are separated
- See how left alignment is preserved
- See how all the existing mechanisms are still working nicely

**Comment by aadcg** [comment]
@lansingthomas, if you're advocating for constant body indentation in each section (regardless of its nestedness), then I am 100% for. Exemplified below:


```
# 1. Title
....Lorem ipsum

## 1.1 Title
....Loprem ipsum

### 1.1.1 Title
....Lorem ipsum

# 2. Title
....Lorem ipsum
```

If you're advocating for what I exemplify below, I have some doubts! 


```
# 1. Title
....Lorem ipsum

.## 1.1 Title
.....Loprem ipsum

..### 1.1.1 Title
......Lorem ipsum

# 2. Title
....Lorem ipsum
```

Which of these do you find best?

**Comment by lansingthomas** [comment]
preserving nested hierarchy and left alignment

**Comment by aadcg** [comment]
We're absolutely on the same page. If that was your original proposal from the beginning, then I'm sorry! I'm afraid I have misinterpreted it...

**Comment by lansingthomas** [comment]
for the tutorial and the manual

**Comment by lansingthomas** [comment]
haha no worries -- im glad you commented now -- I have a whole freaking three paragraphs of blahblahblah about this written out in another window. 
we are on the same page!

**Comment by lansingthomas** [comment]
Vote to indent body content in `tutorial` and `manual`
- subheadings
- lists
- paragraphs
same indentation for all (preserving what exists)

Myself
- [x] votes yes
- [ ] votes no

@jmercouris
- [ ] votes yes
- [ ] votes no

@Ambrevar 
- [ ] votes yes
- [ ] votes no

@aadcg 
- [ ] votes yes
- [ ] votes no

@aartaka 
- [ ] votes yes
- [ ] votes no

**Comment by lansingthomas** [comment]
@jmercouris please do this :D

artyom and I re-arrived at this layout a few days ago during our typography conversation... I come to post it and look here is an issue already agreed to : )

**Comment by lansingthomas** [comment]
So it should be like this:
<img width="720" alt="manual page alignment (3)" src="https://github.com/atlas-engineer/nyxt/assets/6666079/81a38dee-15fd-41bc-966d-47b5a74c9d60">

**Comment by jmercouris** [comment]
Got it!

**Comment by jmercouris** [comment]
![Screenshot_20230909_170021](https://github.com/atlas-engineer/nyxt/assets/1691662/77d9b626-677d-4cc9-b764-73c6615b824c)

**Comment by jmercouris** [comment]
What say ye @lansingthomas ?

**Comment by jmercouris** [comment]
![Screenshot_20230909_171138](https://github.com/atlas-engineer/nyxt/assets/1691662/271f26e0-6dbf-4904-94eb-730a4a6be7ed)

a few more tweaks.

**Comment by lansingthomas** [comment]
@jmercouris Looks grand -- thank you ^^
- indentations
   - are
      - awesome!

**Comment by jmercouris** [comment]
Indeed, they aid in readability I think.
